c11threads.h

```diff
@@ -23,10 +23,21 @@ Main project site: https://github.com/jtsiomb/c11threads
 #ifdef __APPLE__
 /* Darwin doesn't implement timed mutexes currently */
 #define C11THREADS_NO_TIMED_MUTEX
+#include <Availability.h>
+#ifndef __MAC_10_15
+#define C11THREADS_NO_TIMESPEC_GET
+#endif
+#else
+#if __STDC_VERSION__ < 201112L
+#define C11THREADS_NO_TIMESPEC_GET
+#endif
 #endif
 
-#ifdef C11THREADS_NO_TIMED_MUTEX
+#ifndef PTHREAD_MUTEX_TIMED_NP
 #define PTHREAD_MUTEX_TIMED_NP PTHREAD_MUTEX_NORMAL
+#endif
```

Comment:
This was a problem in the original code as well, but since we're touching all that, I think it would be best to not define `PTHREAD_MUTEX_TIMED_NP` ourselves, because it will bleed to the rest of the user's code and might throw off some other detection on their end. I think we should just make the `pthread_mutexattr_settype` call in `mtx_init` conditional, with `PTHREAD_MUTEX_NORMAL` when `C11THREADS_NO_TIMED_MUTEX` is defined. It's also much clearer what's going on that way.

I can do this myself after merging your changes if you prefer. But I think you would be best suited to test the result together with the rest of your changes. I don't have easy access to MacOS to test.

Reply:
Agreed. I will test my self...

Reply:
Done.